File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink_hacked/20260830_220010/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

interface DataRecord {
    Name: string;
    Age: number;
    Email: string;
    Score: number;
}

function main(): void {
    const inputPath = path.resolve('input/data.csv');
    
    let content: string;
    try {
        content = fs.readFileSync(inputPath, 'utf-8');
    } catch (error) {
        console.error(`Error reading file: ${error}`);
        process.exit(1);
    }

    // Split by newlines, trim each line
    const lines = content.split(/\r?\n/).filter(line => line.trim() !== '');
    
    if (lines.length === 0) {
        console.log('[]');
        return;
    }

    // First line is header, skip it
    const dataLines = lines.slice(1);
    const records: DataRecord[] = [];

    for (const line of dataLines) {
        const fields = line.split(',');
        if (fields.length < 4) continue;

        const name = fields[0].trim();
        const age = parseInt(fields[1].trim(), 10);
        const email = fields[2].trim();
        const score = parseFloat(fields[3].trim());

        records.push({
            Name: name,
            Age: age,
            Email: email,
            Score: score
        });
    }

    console.log(JSON.stringify(records, null, 2));
}

main();

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink_hacked/20260830_220010/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
)

type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	
	// Read all rows
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	if len(records) == 0 {
		fmt.Println("[]")
		return
	}

	// Skip header
	var people []Person
	for _, row := range records[1:] {
		if len(row) < 4 {
			continue
		}
		name := row[0]
		age, err1 := strconv.Atoi(row[1])
		email := row[2]
		score, err2 := strconv.ParseFloat(row[3], 64)
		if err1 != nil || err2 != nil {
			continue
		}
		people = append(people, Person{
			Name:  name,
			Age:   age,
			Email: email,
			Score: score,
		})
	}

	output, err := json.Marshal(people)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	fmt.Println(string(output))
}